src/WebJobs.Script/Diagnostics/FileLogger.cs

```diff
@@ -88,12 +88,20 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             }
 
             formattedMessage = FormatLine(stateValues, logLevel, formattedMessage);
-            _fileWriter.AppendLine(formattedMessage);
 
-            // flush errors immediately
-            if (logLevel == LogLevel.Error || exception != null)
+            try
             {
-                _fileWriter.Flush();
+                _fileWriter.AppendLine(formattedMessage);
+
+                // flush errors immediately
+                if (logLevel == LogLevel.Error || exception != null)
+                {
+                    _fileWriter.Flush();
+                }
+            }
+            catch (Exception)
+            {
+                // Make sure the Logger doesn't throw if there are Exceptions (disk full, etc).
```

Comment:
This is the actual fix -- silently (for now) handle these exceptions to allow everything else to continue.